### PR TITLE
Fix the issue #559

### PR DESCRIPTION
**Pull request by OussamaBeng**
Fixing the errors logging

### CODE REVIEW

**Comment by Qwarctick** [comment]
Note: This PR is related to https://github.com/wapiti-scanner/wapiti/issues/559

I think this does not work as expected. If I try your code : 

``` bash
bin/wapiti --update --wapp-url qwerty

 ██╗    ██╗ █████╗ ██████╗ ██╗████████╗██╗██████╗
 ██║    ██║██╔══██╗██╔══██╗██║╚══██╔══╝██║╚════██╗
 ██║ █╗ ██║███████║██████╔╝██║   ██║   ██║ █████╔╝
 ██║███╗██║██╔══██║██╔═══╝ ██║   ██║   ██║ ╚═══██╗
 ╚███╔███╔╝██║  ██║██║     ██║   ██║   ██║██████╔╝
  ╚══╝╚══╝ ╚═╝  ╚═╝╚═╝     ╚═╝   ╚═╝   ╚═╝╚═════╝
Wapiti 3.1.8 (wapiti-scanner.github.io)
[*] Updating modules
Updating module htp
Update done.
Updating module wapp
Error: Non-200 status code for _.json. Skipping.
RequestError occurred: Request URL is missing an 'http://' or 'https://' protocol.
RequestError during update: Request URL is missing an 'http://' or 'https://' protocol.
[!] Module wapp seems broken and will be skipped
Updating module nikto
Update done.
Error: None is not a valid domain name
```
You should have one or two error message, not duplicate of them. And there is a mix of error message.

**Review by bretfourbe — APPROVED** [review]

**Comment by Qwarctick** [comment]
Seems good:

``` python
$ bin/wapiti --update --wapp-url kjhdsdkfhdsf

 ██╗    ██╗ █████╗ ██████╗ ██╗████████╗██╗██████╗
 ██║    ██║██╔══██╗██╔══██╗██║╚══██╔══╝██║╚════██╗
 ██║ █╗ ██║███████║██████╔╝██║   ██║   ██║ █████╔╝
 ██║███╗██║██╔══██║██╔═══╝ ██║   ██║   ██║ ╚═══██╗
 ╚███╔███╔╝██║  ██║██║     ██║   ██║   ██║██████╔╝
  ╚══╝╚══╝ ╚═╝  ╚═╝╚═╝     ╚═╝   ╚═╝   ╚═╝╚═════╝
Wapiti 3.1.8 (wapiti-scanner.github.io)
[*] Updating modules
Updating module wapp
Error: kjhdsdkfhdsf is not a valid URL
Invalid Option Error :
[!] Module wapp seems broken and will be skipped
Updating module nikto
Update done.
Updating module htp
Update done.
Error: None is not a valid domain name
```

**Review by Qwarctick — APPROVED** [review]

File: wapitiCore/controller/wapiti.py [hunk 1]
```diff
@@ -335,15 +335,26 @@ async def update(self, requested_modules: str = "all"):
...
                         logging.info(f"Updating module {mod_name}")
-                        await class_instance.update()
+                        try:
+                            await class_instance.update()
+                            logging.success("Update done.")
+                        except RequestError:
+                            logging.error("Request Error :")
+                            raise
+                        except InvalidOptionValue:
+                            logging.error("Invalid Option Error :")
```
Comment (Qwarctick) [inline wapitiCore/controller/wapiti.py:345]: You can print the error directly from the exception as `InvalidOptionValue` implements `__str__`.

``` python
except InvalidOptionValue as invalid_option_value:
    logging.error(invalid_option_value)
```

File: wapitiCore/controller/wapiti.py [hunk 2]
```diff
@@ -335,15 +335,26 @@ async def update(self, requested_modules: str = "all"):
                     )
                     if hasattr(class_instance, "update"):
                         logging.info(f"Updating module {mod_name}")
-                        await class_instance.update()
+                        try:
+                            await class_instance.update()
+                            logging.success("Update done.")
+                        except RequestError:
+                            logging.error("Request Error :")
```
Comment (Qwarctick) [inline wapitiCore/controller/wapiti.py:342]: You can print the error directly from the exception as `InvalidOptionValue` implements `__str__`.

``` python
except RequestError as request_error:
    logging.error(request_error)
```

File: wapitiCore/attack/mod_wapp.py [hunk 3]
```diff
@@ -110,15 +112,24 @@ async def update(self):
...
+                "--wapp-url", self.BASE_URL
+            )
         try:
             await self._load_wapp_database(
                 wapp_categories_url,
                 wapp_technologies_base_url,
                 wapp_groups_url
             )
+        except RequestError as e:
+            logging.error(f"RequestError occurred: {e}")
```
Comment (Qwarctick) [inline wapitiCore/attack/mod_wapp.py:126]: You should just raise again the error. Remove the logging to avoid duplication.

**Review by Qwarctick — CHANGES_REQUESTED** [review]

**Pull request merged**